src/Tizen.NUI/src/public/Common/Vector4.cs

```diff
@@ -39,6 +75,21 @@ public class Vector4 : Disposable, ICloneable
             // Do nothing. Just call for load static values.
         }
 
+        private static DaliVector4 singletonObj;
+
+        internal static DaliVector4 SingletonDaliVector4
```

Comment:
Minor, but important questoin for this concept.

Since `Color` (and `Vector4` also I think) has no main thread dependency, we could make and change them at workerthread freely. (For example, user try to build `PropertyMap` at worker thread, and try to call `PropertyValue.Get(Color)`).

Is current implement thread safe enough? I hope we `must` consider multi thread cases for this `SingletonDaliVector4` using scenario.

Reply:
This type is internal, so it can only be used by NUI,
I have checked the usage of Color, except the PropertyValue,
all value of Color will be set to some Properties of View,
so that it will be done in main thread.

About the PropertyValue, I will use the "NewPropertyValueVector4Componentwise" to instead of "NewPropertyValueVector4", so that is will not visit the SwigCPtr of Color.

Reply:
Looks fine to me now. For now, I think their is no more usage to get / set Color class's value from Dali at workerthread, in Tizen.NUI side.

Now it looks work wel.